lib/galaxy/webapps/galaxy/controllers/history.py

```diff
@@ -184,28 +184,29 @@ def make_private(self, trans, history_id=None, all_histories=False, **kwd):
             if history:
                 histories.append(history)
         if not histories:
-            return self.message_exception(trans, "Invalid history or histories specified.")
+            raise exceptions.RequestParameterMissingException("No history or histories specified.")
         private_role = trans.app.security_agent.get_private_user_role(trans.user)
         user_roles = trans.user.all_roles()
         private_permissions = {
             trans.app.security_agent.permitted_actions.DATASET_MANAGE_PERMISSIONS: [private_role],
             trans.app.security_agent.permitted_actions.DATASET_ACCESS: [private_role],
         }
         for history in histories:
-            self.history_manager.error_unless_mutable(history)
-            # Set default role for history to private
-            trans.app.security_agent.history_set_default_permissions(history, private_permissions)
-            # Set private role for all datasets
-            for hda in history.datasets:
-                if (
-                    not hda.dataset.library_associations
-                    and not trans.app.security_agent.dataset_is_private_to_user(trans, hda.dataset)
-                    and trans.app.security_agent.can_manage_dataset(user_roles, hda.dataset)
-                ):
-                    # If it's not private to me, and I can manage it, set fixed private permissions.
-                    trans.app.security_agent.set_all_dataset_permissions(hda.dataset, private_permissions)
-                    if not trans.app.security_agent.dataset_is_private_to_user(trans, hda.dataset):
-                        raise exceptions.InternalServerError("An error occurred and the dataset is NOT private.")
+            try:
+                # Set default role for history to private
+                trans.app.security_agent.history_set_default_permissions(history, private_permissions)
+                # Set private role for all datasets
+                for hda in history.datasets:
+                    if (
+                        not hda.dataset.library_associations
+                        and not trans.app.security_agent.dataset_is_private_to_user(trans, hda.dataset)
+                        and trans.app.security_agent.can_manage_dataset(user_roles, hda.dataset)
+                    ):
+                        # If it's not private to me, and I can manage it, set fixed private permissions.
+                        trans.app.security_agent.set_all_dataset_permissions(hda.dataset, private_permissions)
+            except Exception:
```

Comment:
so then we can also remove this. if we ever hit this we would lie to the user about the success of the operation 😆

Reply:
(and it would also swallow any underlying message exception ... not that there are any now, but better to avoid this problem now)

Reply:
Ahh alright! makes sense!